src/Tools/dotnet-monitor/ConfigurationJsonWriter.cs

```diff
@@ -230,12 +232,22 @@ private void ProcessSection(IConfigurationSection section, bool includeChildSect
                     ProcessChildren(section, includeChildSections, redact);
                 }
             }
-            else if (!children.Any())
+            else if (!hasValidChildren)
             {
                 WriteValue(section.Value, redact);
             }
         }
 
+        private bool SectionHasValidChildren(IConfigurationSection section, IEnumerable<IConfigurationSection> children)
+        {
+            if (section.Key.Equals(nameof(RootOptions.DiagnosticPort)))
```

Comment:
```suggestion
            if (section.Path.Equals(nameof(RootOptions.DiagnosticPort)))
```

To make sure it is only filtering on based on the rooted path named "DiagnosticPort" rather than accidentally considering some other configuration section that happens to have a key of "DiagnosticPort".